Tasks/Gulp/gulptask.ts

```diff
@@ -2,118 +2,152 @@
-var isCodeCoverageEnabled = tl.getBoolInput('enableCodeCoverage');
-var publishJUnitResults = tl.getBoolInput('publishJUnitResults');
-var testResultsFiles = tl.getInput('testResultsFiles', publishJUnitResults);
-var cwd = tl.getPathInput('cwd', true, false);
-tl.mkdirP(cwd);
-tl.cd(cwd);
-
-tl.debug('check path : ' + gulp);
-if (!tl.exist(gulp)) {
-	tl.debug('not found global installed gulp, try to find gulp locally.');
-	var gt = tl.createToolRunner(tl.which('node', true));
-	var gulpjs = tl.getInput('gulpjs', true);
-	gulpjs = path.resolve(cwd, gulpjs);
-	tl.debug('check path : ' + gulpjs);
-	if (!tl.exist(gulpjs)) {
-		tl.setResult(tl.TaskResult.Failed, tl.loc('GulpNotInstalled', gulpjs));
-	}
-	gt.pathArg(gulpjs);
-}
-else {
-	var gt = tl.createToolRunner(gulp);
-}
-
-if (isCodeCoverageEnabled) {
-	var npm = tl.createToolRunner(tl.which('npm', true));
-	npm.argString('install istanbul');
-	var testFramework = tl.getInput('testFramework', true);
-	var srcFiles = tl.getInput('srcFiles', false);
-	var testSrc = tl.getPathInput('testFiles', true, false);
-	var istanbul = tl.createToolRunner(tl.which('node', true));
-	istanbul.arg('./node_modules/istanbul/lib/cli.js');
-	istanbul.argString('cover --report cobertura --report html');
-	if (srcFiles) {
-		istanbul.argString('-i .' + path.sep + path.join(srcFiles));
+	tl.debug('check path : ' + gulp);
+	if (!tl.exist(gulp)) {
+		tl.debug('not found global installed gulp, try to find gulp locally.');
+		var gt = tl.tool(tl.which('node', true));
+		var gulpjs = tl.getInput('gulpjs', true);
+		gulpjs = path.resolve(cwd, gulpjs);
+		tl.debug('check path : ' + gulpjs);
+		if (!tl.exist(gulpjs)) {
+			tl.setResult(tl.TaskResult.Failed, tl.loc('GulpNotInstalled', gulpjs));
+			return;
+		}
+		gt.arg(gulpjs);
 	}
-	if (testFramework.toLowerCase() == 'jasmine') {
-		istanbul.argString('./node_modules/jasmine/bin/jasmine.js JASMINE_CONFIG_PATH=node_modules/jasmine/lib/examples/jasmine.json');
-	} else {
-		istanbul.arg('./node_modules/mocha/bin/_mocha');
+	else {
+		var gt = tl.tool(gulp);
 	}
-	istanbul.arg(testSrc);
-	var summaryFile = path.join(cwd, 'coverage/cobertura-coverage.xml');
-	var reportDirectory = path.join(cwd, 'coverage/');
-}
 
-// optional - no targets will concat nothing
-gt.arg(tl.getDelimitedInput('targets', ' ', false));
-gt.arg('--gulpfile');
-gt.pathArg(gulpFile);
-gt.argString(tl.getInput('arguments', false));
-gt.exec().then(function (code) {
-	publishTestResults(publishJUnitResults, testResultsFiles);
 	if (isCodeCoverageEnabled) {
-		npm.exec().then(function () {
-			istanbul.exec().then(function (code) {
-				publishCodeCoverage(summaryFile);
-				tl.setResult(tl.TaskResult.Succeeded, tl.loc('GulpReturnCode', code));
-			}).fail(function (err) {
-				publishCodeCoverage(summaryFile);
-				tl.debug('taskRunner fail');
-				tl.setResult(tl.TaskResult.Failed, tl.loc('IstanbulFailed', err.message));
-			});
-		}).fail(function (err) {
+		var npm = tl.tool(tl.which('npm', true));
```

Comment:
i think the plan is to remove this code coverage block altogether, since that belongs in a separate task? if so, now would be the right time to do that (and bump major version),